subsystems/TypeHandlerLibrary/src/test/java/org/terasology/persistence/typeHandling/coreTypes/factories/ObjectFieldMapTypeHandlerFactoryTest.java

```diff
@@ -30,22 +35,81 @@ public class ObjectFieldMapTypeHandlerFactoryTest {
     private final TypeHandlerContext context =
             new TypeHandlerContext(typeHandlerLibrary, mock(SerializationSandbox.class));
 
-    private static class SomeClass<T> {
-        private T t;
-        private List<T> list;
+    @Test
+    @DisplayName("Test that type handler is correctly created via type handler factory")
+    public void testSingleTypeClassTypeHandlerCreationViaFactory() {
+        Optional<TypeHandler<SingleTypeClass<Integer>>> typeHandler =
+                typeHandlerFactory.create(new TypeInfo<SingleTypeClass<Integer>>() { }, context);
+
+        assertTrue(typeHandler.isPresent());
+        assertTrue(typeHandler.get() instanceof ObjectFieldMapTypeHandler);
     }
 
     @Test
-    public void testObject() {
-        Optional<TypeHandler<SomeClass<Integer>>> typeHandler =
-                typeHandlerFactory.create(new TypeInfo<SomeClass<Integer>>() { }, context);
+    @DisplayName("Test that type handler is correctly created via type handler factory")
+    public void testMultiTypeClassTypeHandlerCreationViaFactory() {
+        Optional<TypeHandler<MultiTypeClass<Integer, List<Integer>>>> typeHandler =
+                typeHandlerFactory.create(new TypeInfo<MultiTypeClass<Integer, List<Integer>>>() { }, context);
 
         assertTrue(typeHandler.isPresent());
         assertTrue(typeHandler.get() instanceof ObjectFieldMapTypeHandler);
+    }
 
-        // Verify that the Integer and List<Integer> TypeHandlers were loaded from the TypeHandlerLibrary
-        verify(typeHandlerLibrary).getTypeHandler(eq(TypeInfo.of(Integer.class).getType()));
+    @Test
+    @DisplayName("Test implicit type handler loading for plain old java objects (pojo)")
+    public void testPojo() {
+        typeHandlerFactory.create(new TypeInfo<SingleTypeClass<Integer>>() { }, context);
+
+        // Verify that the Integer TypeHandler loading was called on the TypeHandlerLibrary
+        verify(typeHandlerLibrary, times(1)).getTypeHandler((Type) any());
+        verify(typeHandlerLibrary, times(1)).getTypeHandler(eq(TypeInfo.of(Integer.class).getType()));
+        verify(typeHandlerLibrary, never()).getTypeHandler((Class<Object>) any());
+        verify(typeHandlerLibrary, never()).getTypeHandler((TypeInfo<Object>) any());
```

Comment:
I'd like to explain what we are doing here at least once.

Also, I'd move the specific test up (test happy path) and then make the "negative tests" afterwards.

```suggestion
	  // verify that the expected typehandler was created
        verify(typeHandlerLibrary, times(1)).getTypeHandler(eq(TypeInfo.of(Integer.class).getType()));
        
        // verify that creation was triggered only for the expected typehandler, but not for others
        // 'getTypeHandler'  can be called with either Type, Class, or TypeInfo, and we only expect one call with Type
        verify(typeHandlerLibrary, times(1)).getTypeHandler((Type) any());
        verify(typeHandlerLibrary, never()).getTypeHandler((Class<Object>) any());
        verify(typeHandlerLibrary, never()).getTypeHandler((TypeInfo<Object>) any());
```